src/components/Gallery/Gallery.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import Link from '../Link';
+import Image from '../Image/Image';
+
+const GalleryContainer = styled.div`
+  width: ${(props) => `calc((${props.initiallyShow} * ${props.width}) + .4rem)`};
+  height: ${(props) => `calc(${props.height} + .4rem)`};
+  overflow-x: ${(props) => props.scrollDirection === 'horizontal' ? 'scroll' : 'hidden'};
+  overflow-y: ${(props) => props.scrollDirection === 'vertical' ? 'scroll' : 'hidden'};
+  white-space: ${(props) => props.scrollDirection === 'horizontal' ? 'nowrap' : 'initial'};
+`;
+
+const ImageContainer = styled.span`
+  width: ${(props) => `calc(${props.width} + .4rem)`};
+  height: ${(props) => `calc(${props.height} + .4rem)`};
+  border: 0.4rem solid #f00;
+  display: inline-block;
+`;
+
+const Gallery = ({ width, height, images, isAvailable, initiallyShow, scrollDirection }) => (
+
+  <GalleryContainer
+    width={width}
+    height={height}
+    initiallyShow={initiallyShow}
+    scrollDirection={scrollDirection}
+  >
+    {
+      images.length ? images.map((image) => (
+
+        <ImageContainer
+          width={width}
+          height={height}
+        >
+          <Link
```

Comment:
Link should not be part of Gallery component. Think of it in this way:
Gallery is a component which holds collection of images, so the link becomes part of image ( composition of components ).